libcperciva/util/readpass.c

```diff
@@ -38,10 +38,11 @@ handle(int sig)
 	gotsig[sig] = 1;
 }
 
+/* Restore old signals and re-issue intercepted signals. */
 static void
 resetsigs(struct sigaction savedsa[NSIGS])
 {
-	int i;
+	size_t i;
```

Comment:
Not 100% certain you want this, but it seems to match the rest of the code better.